daemon/container.c

```diff
@@ -2453,9 +2467,31 @@ char *
 container_get_usbtoken_serial(const container_t *container)
 {
 	ASSERT(container);
+	ASSERT(CONTAINER_TOKEN_TYPE_USB == container->token.type);
+
 	return container->token.serial;
 }
 
+char *
+container_get_usbtoken_devpath(const container_t *container)
+{
+	ASSERT(container);
+	ASSERT(CONTAINER_TOKEN_TYPE_USB == container->token.type);
+
+	return container->token.devpath;
+}
+
+void
+container_set_usbtoken_devpath(container_t *container, char *devpath)
+{
+	ASSERT(container);
+	ASSERT(CONTAINER_TOKEN_TYPE_USB == container->token.type);
+
+	DEBUG("Setting token devpath for container %s to %s", container->name, devpath);
+
+	container->token.devpath = devpath;
```

Comment:
memory ownership is unclear to me. i believe devpath should be strdup'ed here

Reply:
done